src/projects/bestfirst.clj

```diff
@@ -14,41 +14,38 @@
   between two locations through many locations
   with different possible costs - using the best first search algorithm"
   ([state goal lmg map]
-   (best-firstSearchA state goal lmg map []))
-  ([state goal lmg map been]
+   (best-firstSearchA state goal lmg map [] []))
+
+  ([state goal lmg map been finalRoute]
    (cond
+     (not (containsLocation map goal))
+       (str "this map does not contain that location")
```

Comment:
I like this addition, perhaps we should be validating to input state as well as goal.

but this can be done in another PR